control-plane/api-gateway/common/filtererrors.go

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package common
+
+import (
+	"errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ErrInvalidExternalRefType = errors.New("invalid kind")
+var ErrExternalRefNotFound = errors.New("ref not found")
+
+type HTTPRouteFilterError struct {
+	Kind    error
+	Message string
+	client.Object
```

Comment:
This makes it so that the httproutefilter error can be passed around our other functions along with the filters themselves, so they can be sent back to the user during bindRoute.

Reply:
I'm not entirely sure I'm following here? Why not just follow the pattern here?

https://github.com/hashicorp/consul-k8s/blob/0100fa4f066a184de8ae11976ef70b5afa85fe39/control-plane/api-gateway/binding/route_binding.go#L81-L89

Why the need to pass errors around outside of the validation result?

Reply:
The issue I was trying to solve for was that the error was occurring when the route was being pulled and I wasn't sure how to account for that, but I think I actually see an easier way to deal with that.

Reply:
No longer using this pattern